### DISCUSSION TITLE
Add search select

### DISCUSSION THREAD

**Discussion by benjivm** [tailwindlabs/headlessui#537] [Ideas]
Hello, I looked for existing issues and on the main website but didn't find anything related to the possibility if a search select component: [something like this](https://react-select-search.com/?path=/story/single-select--search) would be a valuable add to this library.

**Comment by sathovsepyan** [comment]
I tried implementing this by adding an `input` in `Listbox.Button` and it would work fine, except after the click on the input the focus automatically goes to `Listbox.Options`.

Here is the  [codesandbox](https://codesandbox.io/s/condescending-pare-kwmnt?file=/src/App.js) with my attempts.

I'm wondering if it's possible to avoid this focus change and have search in Listbox.

**Comment by benjivm** [comment]
[Thank you!](https://headlessui.dev/react/combobox)
